letsencrypt/tests/test_letsencrypt.py

```diff
@@ -1,14 +1,65 @@
 # -*- coding: utf-8 -*-
-# © 2016 Therp BV <http://therp.nl>
+# Copyright 2018 Therp BV <http://therp.nl>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-from odoo.tests.common import TransactionCase
+from odoo.tests import SingleTransactionCase
+import mock
 
 
-class TestLetsencrypt(TransactionCase):
-    def test_letsencrypt(self):
-        from ..hooks import post_init_hook
-        post_init_hook(self.cr, None)
-        self.env.ref('letsencrypt.config_parameter_reload').write({
-            'value': '',
-        })
-        self.env['letsencrypt'].with_context(letsencrypt_dry_run=True).cron()
+class Mock(mock.MagicMock):
+
+    def __getattr__(self, name):
+        return super(Mock, self).__getattr__(name)
+
+
+class TestLetsencrypt(SingleTransactionCase):
+
+    post_install = True
+
+    def setUp(self):
+        super(TestLetsencrypt, self).setUp()
+
+    def test_config_settings(self):
+        settings_model = self.env['base.config.settings']
+        settings = settings_model.create({
+            'dns_provider': 'script',
+            'script_path': '/a/path/',
+            'altnames': '*.test.com,test.test.*.test.com',
+            'reload_command': 'echo',
+            'demo': 'True',
+            })
+        settings.set_dns_provider()
+        setting_vals = settings.default_get([])
+        self.assertEquals(setting_vals['dns_provider'], 'script')
+        self.assertEquals(setting_vals['script_path'], '/a/path/')
+        self.assertEquals(
+            setting_vals['altnames'],
+            '*.test.com,test.test.*.test.com',
+        )
+        self.assertEquals(setting_vals['reload_command'], 'echo')
+        settings.onchange_altnames()
+        self.assertEquals(settings.needs_dns_provider, True)
+        self.assertEquals(bool(settings.demo), True)
+
+    @mock.patch('odoo.addons.letsencrypt.models.letsencrypt.client')
+    def test_letsencrypt(self, mock_obj):
```

Comment:
@hbrunn This is my first usage of this library and for some reason I think this can be done in a better way. It works but does not look good. Any suggestions welcomed.

Reply:
look good, you just need to assert stuff to check this actually did what you want.

For callables, you can assign a Mock object too instead of a lambda (assign it to this Mock's side_effect), then you can do stuff like `.assert_called_with` to be sure your code calls the library the way it should.

And please don't continuously rebase during development, this makes it hard to review just the delta.